source/extensions/filters/network/thrift_proxy/conn_manager.h

```diff
@@ -114,7 +115,7 @@ class ConnectionManager : public Network::ReadFilter,
                      public ThriftFilters::FilterChainFactoryCallbacks {
     ActiveRpc(ConnectionManager& parent)
         : parent_(parent), request_timer_(new Stats::Timespan(parent_.stats_.request_time_ms_)),
-          stream_id_(parent_.stream_id_++) {
+          stream_id_(parent_.random_generator_.random()) {
```

Comment:
If two streams active at the same time have the same randomly generated ID, won't it break the matching of responses to requests?

Reply:
Though this was originally intended to be the sequence id, we were only ever using this to allow ENVOY_STREAM_LOG to work because the upstream sequence IDs actually need to be tracked per-connection. I'm working on another PR that does that.